crates/wallet/src/lib.rs

```diff
@@ -210,82 +364,43 @@ where
         // we acquire the permit here so that all following operations are performed exclusively
         let _permit = self.inner.permit.lock().await;
 
-        // set nonce
-        let next_nonce = LoadState::next_available_nonce(
-            &self.inner.eth_api,
-            NetworkWallet::<Ethereum>::default_signer_address(&self.inner.wallet),
-        )
-        .await
-        .map_err(|err| {
-            self.inner.metrics.invalid_send_transaction_calls.increment(1);
-            err.into()
-        })?;
-        request.nonce = Some(next_nonce);
-
         // set chain id
         request.chain_id = Some(self.chain_id());
 
         // set gas limit
         // note: we also set the `from` field here to correctly estimate for contracts that use e.g.
         // `tx.origin`
-        request.from = Some(NetworkWallet::<Ethereum>::default_signer_address(&self.inner.wallet));
-        let (estimate, base_fee) = tokio::join!(
-            EthCall::estimate_gas_at(&self.inner.eth_api, request.clone(), BlockId::latest(), None),
-            LoadFee::eip1559_fees(&self.inner.eth_api, None, None)
-        );
-        let estimate = estimate.map_err(|err| {
-            self.inner.metrics.invalid_send_transaction_calls.increment(1);
-            err.into()
-        })?;
-
-        if estimate >= U256::from(350_000) {
+        request.from = Some(self.inner.upstream.default_signer_address());
```

Comment:
I guess we technically don't need this if we expect that the type takes care of this.

but I think having it like this is better